src/fiber/types.rs

```diff
@@ -363,6 +363,29 @@ impl Privkey {
         );
         EcdsaSignature::from(sig)
     }
+
+    pub fn x_only_pub_key(&self) -> XOnlyPublicKey {
+        let secp256k1_instance = secp256k1_instance();
+        let secret_key = self.0;
+        let keypair = secp256k1::Keypair::from_secret_key(secp256k1_instance, &secret_key);
+        XOnlyPublicKey::from_keypair(&keypair).0
+    }
+
+    pub fn sign_schnorr(&self, message: [u8; 32]) -> SchnorrSignature {
```

Comment:
I think these two fn are for testing only, please add conditional flag for these and remove the unnecessary trace log.